src/ocean/net/ssl/openssl/OpenSsl.d

```diff
@@ -0,0 +1,833 @@
+
+        Opaque struct which holds RSA encryption parameters
+
+    ***************************************************************************/
+
+    public struct RSA;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a message digest
+
+    ***************************************************************************/
+
+    public struct EVP_MD;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a message digest context
+
+    ***************************************************************************/
+
+    public struct EVP_MD_CTX;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a private key
+
+    ***************************************************************************/
+
+    public struct EVP_PKEY;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a private key context
+
+    ***************************************************************************/
+
+    public struct EVP_PKEY_CTX;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a digest encryption engine
+
+    ***************************************************************************/
+
+    public struct ENGINE;
+
+
+    /***************************************************************************
+
+        Retrieve the function pointers for SSLv3, or v2 if v3 is unavailable
+
+        Returns;
+            A struct with all of the function pointers pointing to the SSL v3
+            functions, falling back to v2 if v3 is unavailable
+
+    ***************************************************************************/
+
+    public ssl_method_st* SSLv23_method();
+
+
+    /***************************************************************************
+
+        Initialize the SSL library by registering algorithms
+
+        Returns:
+            Always returns 1.
+
+    ***************************************************************************/
+
+    public int SSL_library_init ();
+
+
+    /***************************************************************************
+
+        Registers the error strings for all libcrypto and libssl function
+
+    ***************************************************************************/
+
+    public void  SSL_load_error_strings ();
+
+
+    /***************************************************************************
+
+        Obtains a human-readable error message
+
+        Params:
+            e = the error code
+
+        Returns:
+            A human-readable string representing the error code.
+
+    ***************************************************************************/
+
+    public Const!(char *) ERR_reason_error_string (c_ulong e);
```

Comment:
Nit: `Const!(char)*`. But usually we (used to?) go with `Const!(char)*`.
One of the advantage is that it makes `auto` slightly more usable, but not a big deal either way.